esx_service/utils/vmdk_utils.py

```diff
@@ -62,17 +69,31 @@ def get_datastores():
                    os.path.join(d.info.url, 'dockvols'))
                   for d in ds_objects]
     pyVim.connect.Disconnect(si)
+    logging.debug("Disconnect from localhost si %s", si)
 
     return datastores
 
-def get_volumes():
+def get_volumes(tenant_re):
     """ Return dicts of docker volumes, their datastore and their paths """
+    logging.debug("get_volumes: tenant_pattern(%s)", tenant_re)
     volumes = []
     for (datastore, url_name, path) in get_datastores():
-        for file_name in list_vmdks(path):
-            volumes.append({'path': path,
-                            'filename': file_name,
-                            'datastore': datastore})
+        logging.debug("get_volumes: %s %s %s", datastore, url_name, path)
+        if not tenant_re:
+            for file_name in list_vmdks(path):
+                volumes.append({'path': path,
+                                'filename': file_name,
+                                'datastore': datastore})
+        else:
+            for root, dirs, files in os.walk(path):
```

Comment:
can you add a comment (here or in docstring) about what is that you are matching here? The function param is called tenant_re ("re" is regular expression I assume?) but the code below seems to do glob matching (fnmatch) ? Also, os.path shoudl be uded instead of string manipulation (replace and then [1:]. If you still want the strings, please add comment about what filepaths you expect here.